src/Console/Command/GenerateConfigCommand.php

```diff
@@ -4,78 +4,85 @@
 
 final class GenerateConfigCommand extends Command
 {
     /**
-     * @var WorkerInterface[]
+     * @var string
      */
-    private $workers = [];
+    public const NAME = 'craft:generate';
 
     /**
      * @var SymfonyStyle
      */
     private $symfonyStyle;
 
     /**
-     * @param WorkerInterface[] $workers
+     * @var YamlGenerator
      */
-    public function __construct(array $workers, SymfonyStyle $symfonyStyle)
-    {
-        parent::__construct();
-        $this->workers = $workers;
+    private $yamlGenerator;
+
+    /**
+     * @var YamlPrinter
+     */
+    private $yamlPrinter;
+
+    /**
+     * @var JsonReader
+     */
+    private $jsonReader;
+
+    public function __construct(
+        SymfonyStyle $symfonyStyle,
+        JsonReader $jsonReader,
+        YamlGenerator $yamlGenerator,
+        YamlPrinter $yamlPrinter
+    ) {
         $this->symfonyStyle = $symfonyStyle;
+        $this->yamlGenerator = $yamlGenerator;
+        $this->yamlPrinter = $yamlPrinter;
+        $this->jsonReader = $jsonReader;
+
+        parent::__construct();
     }
 
     public function execute(InputInterface $input, OutputInterface $output): int
     {
-        $composerJsonFile = __DIR__ . '/../../../composer.json';
+        $file = $input->getArgument('file');
 
-        $composerJsonContent = Json::decode(FileSystem::read($composerJsonFile), Json::FORCE_ARRAY);
+        // the main application :)
+        // 1. read
+        $composerJsonContent = $this->jsonReader->readFileToJson($file);
 
-        $ciYaml = [];
-        foreach ($this->workers as $worker) {
-            if ($worker->isMatch($composerJsonContent, 'require', 'php')) {
-                $ciYaml['language'] = 'php';
-                $ciYaml['install'] = '- composer install';
-                $ciYaml['script'] = 'skip';
+        // 2. process
+        $ciYaml = $this->yamlGenerator->generateFromComposerJson($composerJsonContent);
 
-                $ciYaml['jobs']['include']['stage'] = 'preparing';
-                $ciYaml['jobs']['include']['name:'] = 'prepare';
-                $ciYaml['jobs']['include']['php'] = $composerJsonContent['require']['php'];
-                $ciYaml['jobs']['include']['script'] = '- do something';
+        // 3. print
+        $outputFile = getcwd() . '/example.yaml';
+        $this->yamlPrinter->printYamlToFile($ciYaml, $outputFile);
 
-                if ($worker->isMatch($composerJsonContent, 'require-dev', 'phpstan/phpstan')) {
-                    $ciYaml['jobs']['include']['stage'] = 'testing';
-                    $ciYaml['jobs']['include']['name'] = 'phpstan/phpstan';
-                    $ciYaml['jobs']['include']['php'] = $composerJsonContent['require']['php'];
-                    $ciYaml['jobs']['include']['script'] = '- composer check-cs';
-                }
-            } else {
-                $ciYaml['language'] = 'other';
-            }
-        }
-
-        $yaml = Yaml::dump($ciYaml, 2, 4, Yaml::DUMP_OBJECT_AS_MAP);
```

Comment:
`YamlPrinter` was decopuled, to prevent ambiguous Yaml dumping like:

```php
$yaml = Yaml::dump($ciYaml, 2, 4, Yaml::DUMP_OBJECT_AS_MAP);
$yaml = Yaml::dump($ciYaml, 10, 4, Yaml::DUMP_OBJECT_AS_MAP);
$yaml = Yaml::dump($ciYaml, 10, 8);
```